gui/wxpython/gui_core/pyedit.py

```diff
@@ -368,6 +384,10 @@ def OnSave(self, event):
         else:
             self.SaveAs()
 
+        self.guiparent.recent_files.addFileToHistory.emit(
+            filename=self.filename,
+        )
```

Comment:
I guess you are asking why not call a member function directly and you are right, this is not better than calling a function directly which is probably what you should use here. Signals are meant to be used in the way that the object/class which defines them is (typically) the (_only_) one calling emit. Users of that class then connect to the signal.

You could do signals both ways, but that brings challenges and the controller is in control and has access to the objects (frame and controller aggregate the objects), so why not call the functions directly.